sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/CosmosQueryRequestOptionsImpl.java

```diff
@@ -235,6 +237,18 @@ public Integer getMaxItemCountForVectorSearch() {
         return this.maxItemCountForVectorSearch;
     }
 
+    /**
+     * Gets the maximum item size to fetch during hybrid search queries.
+     *
+     * @return the max number of items for hybrid search.
+     */
+    public Integer getMaxItemCountForHybridSearch() {
+        if (this.maxItemCountForHybridSearch == null) {
```

Comment:
A getter (getXXX method) shoudl never have any side effects - so changing the underlying variable in the get-method is bad. you can retrun the DEFAULt if teh variable is set to null - but please without side effects.